src/testing/flow_solver.cpp

```diff
@@ -1,35 +1,324 @@
+        const int number_of_columns = data_column_names.size();
+
+        std::getline(FILE, line); // read first line of data
+        
+        // check that there indeed is data to be read
+        if (line.empty()) {
+            pcout << "Error: Table has no data to be read" << std::endl;
+            std::abort();
+        }
+
+        std::vector<double> current_line_values(number_of_columns);
+        while (!line.empty()) {
+            std::string dummy_line = line;
+
+            current_line_values[0] = std::stod(dummy_line,&sz1);
+            for(int i=1; i<number_of_columns; ++i) {
+                dummy_line = dummy_line.substr(sz1);
+                sz1 = 0;
+                current_line_values[i] = std::stod(dummy_line,&sz1);
+            }
+
+            // Add data entries to table
+            for(int i=0; i<number_of_columns; ++i) {
+                data_table->add_value(data_column_names[i], current_line_values[i]);
+                data_table->set_precision(data_column_names[i], 16);
+                data_table->set_scientific(data_column_names[i], true);
+            }
+            std::getline(FILE, line); // read next line
+        }
+    }
+}
+
+template <int dim, int nstate>
+std::string FlowSolver<dim,nstate>::double_to_string(const double value_input) const {
+    // converts a double to a string with full precision
+    std::stringstream ss;
+    ss << std::scientific << std::setprecision(16) << value_input;
+    std::string double_to_string = ss.str();
+    return double_to_string;
 }
 
+template <int dim, int nstate>
+void FlowSolver<dim,nstate>::write_restart_parameter_file(
+    const int restart_index_input,
+    const double constant_time_step_input) const {
+    // write the restart parameter file
+    if(this->mpi_rank==0) {
+        // read a copy of the current parameters file
+        std::ifstream CURRENT_FILE(input_parameters_file_reference_copy_filename);
+        
+        // create write file with appropriate postfix given the restart index input
+        const std::string restart_filename = get_restart_filename_without_extension(restart_index_input)+std::string(".prm");
+        std::ofstream RESTART_FILE(flow_solver_param.restart_files_directory_name + std::string("/") + restart_filename);
+
+        // Lines to identify the subsections in the .prm file
+        /* WARNING: (2) These must be in the order they appear in the .prm file
+         */
+        std::vector<std::string> subsection_line;
+        subsection_line.push_back("subsection ODE solver");
+        subsection_line.push_back("subsection flow_solver");
+        // Number of subsections to change values in
+        int number_of_subsections = subsection_line.size();
+
+
+        /* WARNING: (1) Must put a space before and after each parameter string as done below
+         *          (2) These must be in the order they appear in the .prm file
```

Comment:
What happens if either of those warnings aren't satisfied. Is there a way to detect this and tell the user without the user knowing about this comment in the code?

Reply:
As discussed in chat, those warnings are for future changes. A user of the restart capability won't encounter any issues. No need to fail/warn since someone who changes the code will necessarily read the above warning.